lychee-lib/src/types/basic_auth.rs

```diff
@@ -0,0 +1,174 @@
+use std::str::FromStr;
+
+use headers::{authorization::Basic, Authorization};
+use regex::Regex;
+use serde::Deserialize;
+use serde_with::DeserializeFromStr;
+use thiserror::Error;
+
+#[derive(Copy, Clone, Debug, Error)]
+pub enum BasicAuthCredentialsParseError {
+    #[error("Invalid basic auth credentials syntax")]
+    InvalidSyntax,
+
+    #[error("Missing basic auth password")]
+    MissingPassword,
+
+    #[error("Missing basic auth username")]
+    MissingUsername,
+
+    #[error("Too many values separated by colon. Expected 2, got {0}. Valid form is '<username>:<password>'")]
+    TooManyParts(usize),
+}
```

Comment:
I like that you put the auth-related errors into the respective module. We should do the same with the other errors at some point.

Reply:
Yes, agreed. But I guess this is out-of-scope for this PR.

Reply:
Yes, indeed 👍